src/ClientServerChannel.cc

```diff
@@ -272,15 +268,8 @@ CMD  ClientServerChannel::readCommand() {
     std::cerr << "ERROR: reading of message body failed! Socket not ready." << std::endl;
     return CMD_UNDEF;
   }
-//  LOG_DEBUG << "readCommand message:" << std::endl;
-//  for (size_t i=0; i < *message_size; i++) {
-//    const char c = message_buffer[i];
-//    LOG_DEBUG << std::dec << static_cast<int>(c);
-//        LOG_DEBUG << (((i + 1) % 16 == 0) ? '\n' : ' ');
-//  }
-//  LOG_DEBUG << std::endl;
   if ( *message_size > 0 ) {
-    debug_byte_array ( message_buffer, *message_size );
+    NS_LOG_INFO("message buffer as byte array: " << debug_byte_array ( message_buffer, *message_size ));
```

Comment:
You use NS_LOG_INFO everywhere now.. what about DEBUG log level? All those logs should not be printed in the default case, only when DEBUG level is configured.

Reply:
## Decision Explanation
Before this change every log message in `ClientServerChannel.cc` was streamed to `std::cerr`. So, there was no classification or log level distinction. Therefore I used the same *ns-3 severity class* for all messages. Because the messages generated by `ClientServerChannel` were too verbose for common debugging, I used a log level lower than `LOG_DEBUG`.

### Change Options
Apart from that logging can be deactivated or set to a specific level for every component individually. There are some options:

- rethink severity class for all / some log messages in `ClientServerChannel.cc`
- deactivate logging of component `ClientServerChannel` by default (-> `ns3_federate_config.xml`)

## ns-3 Details
ns-3 defines the *severity classes* in this order:

Severity Class | Meaning
-- | --
LOG_NONE | The default, no logging
LOG_ERROR | Serious error messages only
LOG_WARN | Warning messages
LOG_DEBUG | For use in debugging
LOG_INFO | Informational
LOG_FUNCTION | Function tracing
LOG_LOGIC | Control flow tracing within functions

In addition there are *log levels*, defined as:

Level | Meaning
-- | --
LOG_LEVEL_ERROR | Only LOG_ERROR severity class messages.
LOG_LEVEL_WARN | LOG_WARN and above.
LOG_LEVEL_DEBUG | LOG_DEBUG and above.
LOG_LEVEL_INFO | LOG_INFO and above.
LOG_LEVEL_FUNCTION | LOG_FUNCTION and above.
LOG_LEVEL_LOGIC | LOG_LOGIC and above.
LOG_LEVEL_ALL | All severity classes.
LOG_ALL | Synonym for LOG_LEVEL_ALL

*(see [ns-3 manual](https://www.nsnam.org/docs/manual/html/logging.html) for details)*